### ISSUE TITLE
error in runAutoencoderDenoising

### ISSUE THREAD

**Issue by renyzh5** [theMILOlab/SPATA2#145]
seurat_object <- Seurat::SCTransform(seurat_object, assay = "Spatial", verbose = FALSE)
Seurat::SpatialFeaturePlot(seurat_object, features = c("nCount_Spatial")) 

spata_obj <- 
  SPATA2::asSPATA2(
    object = seurat_object,
    sample_name = "P15T",
    image_name = "image_P15_T", 
    spatial_method = "Visium"
  )

######################### #########################
# all expression matrices before denoising
getExpressionMatrixNames(object = spata_obj)
[1] "scaled"

# active expression matrix before denoising
getActiveMatrixName(object = spata_obj)
16:29:26 Using matrix 'counts'.
[1] "counts"

# denoising your data 
spata_obj <-
  runAutoencoderDenoising(
    object = spata_obj, 
    activation = "selu", 
    bottleneck = 56, 
    epochs = 20, 
    layers = c(128, 64, 32), 
    dropout = 0.1
  )

Checking input for validity.
Error in UseMethod("compile") : 
  no applicable method for 'compile' applied to an object of class "c('keras.src.models.functional.Functional', 'keras.models.functional.Functional', 'keras.src.ops.function.Function', 'keras.ops.function.Function', 'keras.src.models.model.Model', 'keras.models.model.Model', 'keras.src.backend.tensorflow.trainer.TensorFlowTrainer', 'keras.backend.tensorflow.trainer.TensorFlowTrainer', 'keras.src.trainers.trainer.Trainer', 'keras.trainers.trainer.Trainer', 'keras.src.layers.layer.Layer', 'keras.layers.layer.Layer', 'keras.src.backend.tensorflow.layer.TFLayer', 'keras.backend.tensorflow.layer.TFLayer', 'keras.src.backend.tensorflow.trackable.KerasAutoTrackable', 'keras.backend.tensorflow.trackable.KerasAutoTrackable', 'tensorflow.python.trackable.autotrackable.AutoTrackable', 'tensorflow.python.trackable.base.Trackable', 'keras.src.ops.operation.Operation', 'keras.ops.operation.Operation', 'keras.src.saving.keras_saveable.KerasSaveable', 'keras.saving.keras_saveable.KerasSaveable'
In addition: Warning message:
In confuns::give_feedback(msg = msg, fdb.fn = fdb_fn, with.time = FALSE) :
  Argument `of_sample` is deprecated and no longer in use.( used by runAutoencoderDenoising() )
  
 how can I solve this problem? thank you!

**Comment by kueckelj** [comment]
Hello renyzh5, sorry for the late reply. The autoencoder functions are currently under supervision. They will be exported in upcoming version releases again.

**Comment by hlcxq1223** [comment]
> Hello renyzh5, sorry for the late reply. The autoencoder functions are currently under supervision. They will be exported in upcoming version releases again.

Hello kueckelj, did the autoencoder functions remain in the SPATA2 3.0 version? I can't find them in the tutorial.

**Comment by kueckelj** [comment]
You won't find them right now cause I did not export them. I have to adjust the functions to the new version which I have not done yet. It is on my to-do list for the next days but I can not be precise about when I upload them. I will let you know!

**Comment by Jianbo1999** [comment]
What's happen?
X98 <-
+   runAutoencoderDenoising(
+     object = X98, 
+     activation = "selu", 
+     bottleneck = 56, 
+     epochs = 20, 
+     layers = c(128, 64, 32), 
+     dropout = 0.1
+   )
Checking input for validity.
Error in getSampleNames(object) : 
  could not find function "getSampleNames"
